src/lib/core/components/computations/Utils.ts

```diff
@@ -47,3 +50,177 @@ function createRandomString(numChars: number) {
     .toString(36)
     .slice(2, numChars + 2);
 }
+
+export function assertComputationWithConfig<ConfigType>(
+  computation: Computation,
+  decoder: t.Type<Computation, unknown, unknown>
+): asserts computation is Computation<ConfigType> {
+  const onLeft = (errors: t.Errors) => {
```

Comment:
for my education, do you mind explaining a bit about this function? I think it's goal is to assert the correct type, but how does the `onRight`, `onLeft` come into play?

Reply:
So `decode` will return either an object called `left` or an object called `right`. As you could guess, `left` represents an error. The [top-level "Idea" section](https://github.com/gcanti/io-ts/blob/master/index.md#the-idea) of the `io-ts` docs handle it this way. I think it could also be done in a more readable format with the use of `isLeft` and/or `isRight`, along these lines:
```js
if (isLeft(decoder.decode(computation)) throw new Error('bad config')
```